fiatconnect-api.md

```diff
@@ -1710,14 +1713,78 @@ This endpoint MUST fail when the user has no transfer on file with the provided
 
 If the user has no transfer on file with the provided `transferId`, the server MUST return a `ResourceNotFound` error.
 
-# 4. Webhooks
+# 4. Transfer State Machines
+
+The FiatConnect specification requires that both transfers in and out progress through a particular series of states that are made
+available to the client both by polling the `GET /transfer/:transferId/status` endpoint and via webhooks. Statuses for transfers in and
+transfers out must progress according to a specific *state machine*. This state machine is different for both transfers in and transfers out.
+
+## 4.1. Transfers In
+
+Statuses for transfers in must progress through the state machine shown below, ultimately ending at one of the two *terminal* states:
+`TransferFailed` or `TransferComplete`. A server SHOULD make a best effort to reach a terminal transfer state within the `settlementTimeUpperBound`
+specified for the quote associated with the transfer, if one exists. Note that since a transfer in may fail after the server has received fiat funds
+from the user's fiat account but before crypto funds are sent to the user, the server MUST return any fiat funds to the user's fiat account in case
+of such a failure.
+
+```mermaid
+graph LR
+   TransferStarted---->TransferPending
+   TransferPending---->TransferFiatFundsDebited
+   TransferFiatFundsDebited---->TransferWaitingForFiatFunds
+   TransferWaitingForFiatFunds---->TransferReceivedFiatFunds
+   TransferReceivedFiatFunds---->TransferSendingCryptoFunds
+   TransferSendingCryptoFunds---->TransferComplete
+
+   TransferStarted---->TransferFailed
+   TransferPending---->TransferFailed
+   TransferFiatFundsDebited---->TransferFailed
+   TransferWaitingForFiatFunds---->TransferFailed
+   TransferReceivedFiatFunds---->TransferFailed
+   TransferSendingCryptoFunds---->TransferFailed
+```
+
+## 4.2. Transfers Out
+
+Statuses for transfers out must progress through the state machine shown below, ultimately ending at one of the two *terminal* states:
+`TransferFailed` or `TransferComplete`. Note that at the beginning of a transfer out, a server may perform AML checks. If these AML checks
+fail, the server SHOULD progress the state machine to `TransferAmlFailed` before moving to the terminal `TransferFailed` status. A server SHOULD
+make a best effort to reach a terminal transfer state within the `settlementTimeUpperBound` specified for the quote associated with the transfer,
+if one exists.
+
+Transfers out are fundamentally different from transfer in in that they require user interaction partway through the state machine. Once a
+transfer out's state moves to `TransferReadyForUserToSendCryptoFunds`, the server should wait for the user to send the proper amount of crypto
```

Comment:
wait for how long?

Reply:
at least until the quote expires, maybe?

Reply:
yeah, i like waiting until at least the quote expiration here.